### PR TITLE
on `region_errors.rs`

### PR DESCRIPTION
**Pull request by AndyJado**
@rustbot label +A-translation

### CODE REVIEW

**Comment by AndyJado** [comment]
sorry about the mess..but

r?

**Comment by davidtwco** [comment]
You'll need to remove the merge commits, we've got a [no merge policy](https://rustc-dev-guide.rust-lang.org/git.html#no-merge-policy) :)

**Comment by davidtwco** [comment]
I think you've removed the merge commit incorrectly - there are a _lot_ of changes in this pull request now.

**Comment by AndyJado** [comment]
I'll manage to fix it first thing in the morning!

**Review by davidtwco — CHANGES_REQUESTED** [review]
This looks great, I've left a couple comments

**Comment by davidtwco** [comment]
Use `@rustbot ready` when you've addressed the comments :)

File: compiler/rustc_borrowck/src/diagnostics/region_errors.rs [hunk 1]
```diff
@@ -681,39 +676,33 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
             ..
         } = errci;
 
-        let mut diag =
-            self.infcx.tcx.sess.struct_span_err(*span, "lifetime may not live long enough");
-
         let (_, mir_def_name) =
             self.infcx.tcx.article_and_description(self.mir_def_id().to_def_id());
 
+        let err = LifeTimeOutliveErr { span: *span };
```
Comment (davidtwco) [inline compiler/rustc_borrowck/src/diagnostics/region_errors.rs:682]:
```suggestion
        let err = LifetimeOutliveErr { span: *span };
```
Still need to change `LifeTime` to `Lifetime` :)

**Review by davidtwco — COMMENTED** [review]
I think this is ready after this last comment is resolved, thanks for your work on this.

File: compiler/rustc_borrowck/src/diagnostics/region_errors.rs [hunk 2]
```diff
@@ -681,39 +676,33 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
...
+                span: *span,
+                mir_def_name: mir_def_name.to_string(),
+                outlived_fr_name: outlived_fr_name.to_string(),
+                fr_name: fr_name.to_string(),
+            },
+            _ => LifetimeReturnCategoryErr::ShortReturn {
+                span: *span,
+                category_desc: category.description().to_string(),
+                free_region_name: fr_name.to_string(),
+                outlived_fr_name: outlived_fr_name.to_string(),
```
Comment (davidtwco) [inline compiler/rustc_borrowck/src/diagnostics/region_errors.rs:701]: `RegionName` could implement `IntoDiagnosticArg` and be used directly in the diagnostic struct, just calling `into_diagnostic_arg` on its `Symbol`.

**Review by davidtwco — COMMENTED** [review]
Okay, I know that I said that the previous comment was the last one, but I've got one more suggestion - thanks for sticking with this. :)

File: compiler/rustc_borrowck/src/diagnostics/region_errors.rs (hunk 2 above)
Comment (AndyJado) [reply compiler/rustc_borrowck/src/diagnostics/region_errors.rs:701]: every comment you leave, I learn some good stuff. 

And It's okay if you just point out where is `not okay` and then I'll manage to figure it out myself, I'd love that.

**Review by davidtwco — APPROVED** [review]

**Pull request merged**